src/plugins/explore/public/components/visualizations/bar/bucket_options.tsx

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { i18n } from '@osd/i18n';
+import { EuiFormRow, EuiSelect, EuiFieldNumber } from '@elastic/eui';
+import React, { useMemo } from 'react';
+import { useEffectOnce } from 'react-use';
+import { BarChartStyleControls } from './bar_vis_config';
+import { AggregationType, TimeUnit, BucketOptions } from '../types';
+import { getAggregationType, getTimeUnits } from '../utils/collections';
+import { useDebouncedValue } from '../utils/use_debounced_value';
+import { StyleAccordion } from '../style_panel/style_accordion';
+
+interface BucketOptionsPanelProps {
+  styles: BarChartStyleControls['bucket'];
+  bucketType: 'time' | 'num' | 'single' | 'cate';
+  onChange: (styles: BarChartStyleControls['bucket']) => void;
+}
+
+export const BucketOptionsPanel = ({ styles, onChange, bucketType }: BucketOptionsPanelProps) => {
+  const updateBucketOption = <K extends keyof BucketOptions>(key: K, value: BucketOptions[K]) => {
+    onChange({
+      ...styles,
+      [key]: value,
+    });
+  };
+
+  const [localBucketSize, setLocalBucketSize] = useDebouncedValue<number | undefined>(
+    styles?.bucketSize,
+    (val) => updateBucketOption('bucketSize', val)
+  );
+
+  const [localBucketCount, setLocalBucketCount] = useDebouncedValue<number | undefined>(
+    styles?.bucketCount,
+    (val) => updateBucketOption('bucketCount', val)
+  );
+
+  const labelType = useMemo(() => getAggregationType(), []);
+  const timeUnits = useMemo(() => getTimeUnits(), []);
+
+  // initialize old save object with default bucket options
+  useEffectOnce(() => {
+    if (!styles) {
+      const defaultBucket: BucketOptions = {
+        aggregationType: AggregationType.SUM,
+        bucketTimeUnit: TimeUnit.AUTO,
+      };
+      onChange({ ...defaultBucket });
+    }
+  });
+
+  return (
+    <StyleAccordion
+      id="bucketSection"
+      accordionLabel={i18n.translate('explore.stylePanel.bar.tabs.bucket', {
+        defaultMessage: 'Bucket',
+      })}
+      initialIsOpen={true}
+      data-test-subj="bucketPanel"
+    >
+      {bucketType !== 'single' && (
+        <EuiFormRow
+          label={i18n.translate('explore.stylePanel.bar.bucket.type', {
+            defaultMessage: 'Type',
+          })}
+        >
+          <EuiSelect
+            compressed
+            value={styles?.aggregationType}
```

Comment:
Instead of calling `onChange` in `useEffectOnce`, I think better to do this:

```suggestion
            value={styles?.aggregationType ?? defaultBarChartStyles.bucket.aggregationType}
```